stock_reserve/model/stock_reserve.py

```diff
@@ -0,0 +1,197 @@
+              `picking_type_id`.
+        """
+        # if there is 'location_id' field requested, ensure that
+        # picking_type_id is also requested, because it is required
+        # to compute location_id
+        if ('location_id' in fields_list and
+                'picking_type_id' not in fields_list):
+            fields_list = fields_list + ['picking_type_id']
+
+        res = super(StockReservation, self).default_get(fields_list)
+
+        if 'product_qty' in res:
+            del res['product_qty']
+
+        # At this point picking_type_id and location_id
+        # should be computed in default way:
+        #     1. look up context
+        #     2. look up ir_values
+        #     3. look up property fields
+        #     4. look up field.default
+        #     5. delegate to parent model
+        #
+        # If picking_type_id is present and location_id is not, try to find
+        # default value for location_id
+        if not res.get('picking_type_id', None):
+            res['picking_type_id'] = self._default_picking_type_id()
+
+        picking_type_id = res.get('picking_type_id')
+        if picking_type_id and not res.get('location_id', False):
+            picking = self.env['stock.picking'].new(
+                {'picking_type_id': picking_type_id})
+            picking.onchange_picking_type()
+            res['location_id'] = picking.location_id.id
+        if 'location_dest_id' in fields_list:
+            res['location_dest_id'] = self._default_location_dest_id()
+        if 'product_uom_qty' in fields_list:
+            res['product_uom_qty'] = 1.0
+        return res
+
+    @api.model
+    def get_location_from_ref(self, ref):
+        """ Get a location from a xmlid if allowed
+        :param ref: tuple (module, xmlid)
+        """
+        try:
+            location = self.env.ref(ref, raise_if_not_found=True)
+            location.check_access_rule('read')
+            location_id = location.id
+        except (except_orm, ValueError):
+            location_id = False
+        return location_id
+
+    @api.model
+    def _default_picking_type_id(self):
+        ref = 'stock.picking_type_out'
+        return self.env.ref(ref, raise_if_not_found=False).id
+
+    @api.model
+    def _default_location_dest_id(self):
+        ref = 'stock_reserve.stock_location_reservation'
+        return self.get_location_from_ref(ref)
+
+    @api.multi
+    def reserve(self):
+        """ Confirm reservations
+
+        The reservation is done using the default UOM of the product.
+        A date until which the product is reserved can be specified.
+        """
+        self.write({'date_expected': fields.Datetime.now()})
+        self.mapped('move_id')._action_confirm()
+        self.mapped('move_id.picking_id').action_assign()
+        return True
+
+    @api.multi
+    def release(self):
+        """
+        Release moves from reservation
+        """
+        self.mapped('move_id').action_cancel()
```

Comment:
Here you have an error with action_cancel, it's _action_cancel, so you need to add the _